### PR TITLE
Edit button improvements

### PR DESCRIPTION
**Pull request by Zales0123**
Edit button has been added to the order show page. Also, edit button would not be displayed if the order is shipped.

edit-button-no-shipped
edit-button-show

When https://github.com/Setono/sylius-order-edit-plugin/pull/5 is merge, we should introduce a new `OrderInterface::canBeEdited` method to allow easier customisations 🖖

### CODE REVIEW

File: src/Resources/config/services/menu.xml [hunk 1]
```diff
@@ -2,9 +2,9 @@
 <container xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns="http://symfony.com/schema/dic/services"
            xsi:schemaLocation="http://symfony.com/schema/dic/services http://symfony.com/schema/dic/services/services-1.0.xsd">
     <services>
-        <service id="setono_sylius_order_edit.event_subscriber.add_edit_order_action"
-                 class="Setono\SyliusOrderEditPlugin\EventSubscriber\AddEditOrderActionSubscriber">
-            <tag name="kernel.event_subscriber"/>
+        <service id="setono_sylius_order_edit.menu.listener.order_show"
```
Comment (loevgaard) [inline src/Resources/config/services/menu.xml]: Why this change? :D I mean the change from subscriber to listener?

File: src/Resources/config/config.yaml [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+        sylius_admin_order:
+            actions:
+                item:
+                    delete:
+                        type: edit_order
+                        options:
+                            link:
+                                route: sylius_admin_order_update
+                                parameters:
+                                    id: resource.id
```
Comment (loevgaard) [inline src/Resources/config/config.yaml]: I would rather add this in an event subscriber

File: src/Resources/config/config.yaml [hunk 3]
```diff
@@ -0,0 +1,15 @@
+sylius_grid:
+    templates:
+        action:
+            edit_order: "@SetonoSyliusOrderEditPlugin/admin/order/grid/editOrder.html.twig"
```
Comment (loevgaard) [inline src/Resources/config/config.yaml]: I would add this by prepending config

File: src/Resources/config/config.yaml (hunk 2 above)
Comment (Zales0123) [reply src/Resources/config/config.yaml]: I had a question - why you did it with an event subscriber 😄 This way is (of course, it's my opinion) 1000x more readable and straightforward. Also, people in Sylius' ecosystem are used to the yaml configurations. It's also easier to **not** include this change is one decides, as it's simply not importing the configuration file. 

Let's start the discussion :)

Comment (loevgaard) [reply src/Resources/config/config.yaml]: > It's also easier to not include this change is one decides, as it's simply not importing the configuration file

This is only right if this is the only thing you configure in the file. If it was done with an event subscriber, you could allow to disable the event subscriber in the configuration of the plugin _or_ users could remove it in a compiler pass. But, practically, 99.99% would never touch it.

My reasons for event subscriber:
- I _don't_ have to remind people to add a config file
- I like to put as much config inside PHP files because my tooling can help me if anything changes.
- I am able to unit test an event subscriber. With the config approach you actually need a functional test to test that it's added

But it's not that I am against it as I am with other things. I am just trying to move away from including the `config.yaml` that all plugins do when they could instead avoid it. So the argument for me is probably more like: Why would you want people to include a file if they don't have to _and_ you'd get better DX for yourself not doing it?

File: src/Resources/config/services.xml [hunk 4]
```diff
@@ -3,8 +3,8 @@
            xsi:schemaLocation="http://symfony.com/schema/dic/services http://symfony.com/schema/dic/services/services-1.0.xsd">
     <imports>
         <import resource="services/controller.xml"/>
+        <import resource="services/menu.xml"/>
```
Comment (loevgaard) [inline src/Resources/config/services.xml]: I would still have it in the event subscriber namespace, hehe. I know menu is the legacy from Sylius plugins, but I do not agree with it :D

File: src/Resources/config/config.yaml (hunk 2 above)
Comment (Zales0123) [reply src/Resources/config/config.yaml]: I understand your approach. IMO the best compromise is having such configurations in php config files 😄 but I do not thing it's worth to argue more about such a trivial thing, no problem for me to return to the event subscriber 👍

File: src/Resources/config/services.xml (hunk 4 above)
Comment (Zales0123) [reply src/Resources/config/services.xml]: Again, it's probably a matter of taste :dancer: even though the class itself indeed is technically an event subscriber/listener, I feel keeping it in the Menu namespace gives much better information about what _it's responsible for_ rather than _what it is_. Still - let's make it an even subscriber for consistency 👍

Comment (loevgaard) [reply src/Resources/config/services.xml]: I agree on the responsibility argument. From my experience it seems that people most of the times only do this with the menu and all other event subscribers end up in the `EventSubscriber/EventListener` namespace. I find that inconsistent, and that's why I fell back to just having _all_ subscribers in the same namespace

**Review by loevgaard — APPROVED** [review]

**Review by loevgaard — APPROVED** [review]
